staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -228,6 +230,10 @@ func patchResource(
 				if err := strategicPatchObject(codec, defaulter, currentVersionedObject, patchJS, versionedObjToUpdate, versionedObj); err != nil {
 					return nil, err
 				}
+				// ensure the patch didn't change the kind/apiVersion
```

Comment:
Seems like storage validation on updates should have access to check this globally, right?

Reply:
it's only patch that has this issue, right? create/update check the data in the object before getting to storage.

Reply:
> create/update check the data in the object before getting to storage.

If update checked, then I I'm surprised by this since I thought we used the normal update storage to back the patch storage.

Reply:
@deads2k it looks like both PatchResource and UpdateResource are backed by admission.Update, but UpdateResource has [this line](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go#L74)

Reply:
and it looks like there was a [plan](https://github.com/liggitt/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go#L44) to make patch use GuaranteedUpdate